### DISCUSSION TITLE
Infra Setup : On a Cloud ( AWS)

### DISCUSSION THREAD

**Discussion by vipinvijay1986** [egovernments/Digit-Core#36] [Announcements]
Hi Team, 
  I am trying to use [AWS setup for installation.](https://docs.digit.org/setup/quickstart/1.-infra-setup) 
  I am running step 8 of Prerequisite I am facing 22 timeout error.  Do I need to enable any network port for connection? 
 please check the logs below : 
```
awsSetup> ssh -i ./ssh_key ubuntu@13.126.81.228
ssh: connect to host 13.126.81.228 port 22: Connection timed out
awsSetup> ssh -i .\ssh_key.pub ubuntu@13.126.81.228
ssh: connect to host 13.126.81.228 port 22: Connection timed out
```

And I have one more point for ssh key generation I have used the windows utility which generated below two files  (it's it proper or do I need to change anything ) : 

ssh_key
ssh_key.pub

**Comment by gajenc** [comment]
You need to enable the ssh permission in your AWS Instance security group

**Comment by vipinvijay1986** [comment]
@gajendran-egov while doing the  [quickstart profile](https://docs.digit.org/urban/platform/setup-digit/quickstart/1.-choose-infra) installation using AWS VM  ( using my trial account), I faced the following problems : 

1. Prerequisites are not exhaustive for setup and the second step link is not working. 
2. For AWS users, what and all permission is required.  
3. Which AWS profile needs to be configured locally.  

For the 1st and 3rd points, I have added the below documentation but for the 2nd point, I used full privilege(which is not correct). So please update the pages using the below details. 
```
### Prerequisite for setup on **window system using AWS cloud vm**  : 
*For installing the package use powershell administrator mode*. 
1. [Chocolatey](https://chocolatey.org/install#individual), [YouTube reference](https://youtu.be/HxU8CcTUnxA)
5. [Git](https://git-scm.com/download/win)
6. Kubectl command : `choco install kubernetes-cli`
7. curl command : `choco install curl`
8. ssh keygen ( ByDefault in window 11 )  command:  `ssh-keygen` 
9. [Terraform](https://releases.hashicorp.com/terraform/0.14.10/)
10. [AWS cli](https://docs.aws.amazon.com/cli/latest/userguide/getting-started-install.html)
11. [Golang](https://go.dev/dl/) v1.13.3



### Steps : 
* To create a user on AWS console follow the [link](https://www.youtube.com/watch?v=_9u5LuL8KmM) . 
* To create AWSprofile (Access key and Secret key needs to be added from previous step): 
 
aws configure --profile digit-quickstart-poc 
AWS Access Key ID []:
AWS Secret Access Key []:
Default region name []: ap-south-1
Default output format []: text
export AWS_PROFILE=digit-quickstart-poc 
```

**Comment by NeileGov** [comment]
> Hi Team, I am trying to use [AWS setup for installation.](https://docs.digit.org/setup/quickstart/1.-infra-setup) I am running step 8 of Prerequisite I am facing 22 timeout error. Do I need to enable any network port for connection? please check the logs below :
> 
> ```
> awsSetup> ssh -i ./ssh_key ubuntu@13.126.81.228
> ssh: connect to host 13.126.81.228 port 22: Connection timed out
> awsSetup> ssh -i .\ssh_key.pub ubuntu@13.126.81.228
> ssh: connect to host 13.126.81.228 port 22: Connection timed out
> ```
> 
> And I have one more point for ssh key generation I have used the windows utility which generated below two files (it's it proper or do I need to change anything ) :
> 
> ssh_key ssh_key.pub

Could you please go through the below announcement and provide the requisite details?
[https://github.com/egovernments/Digit-Core/discussions/39](url)

**Reply by nikesh2812** [reply]
Yes, you need to open 22, 80, and 6550 ports in your AWS Instance security group. We have mentioned the same in Prerequisites.
